packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -313,30 +328,33 @@ async function filterOutGitignoredFiles(
     return gitIgnoreFilter.filterFiles(files)
 }
 
+export type CollectFilesResultItem = {
+    workspaceFolder: vscode.WorkspaceFolder
+    relativeFilePath: string
+    fileUri: vscode.Uri
+    fileContent: string
+    zipFilePath: string
+}
+export type CollectFilesFileFilter = (relativePath: string) => boolean // returns true if file should be filtered out
+
 /**
- * collects all files that are marked as source
+ * search and collect source files
  * @param sourcePaths the paths where collection starts
  * @param workspaceFolders the current workspace folders opened
- * @param respectGitIgnore whether to respect gitignore file
- * @param addExtraIgnorePatterns whether to add extra exclude patterns even if not in gitignore
+ * @param options - filtering options
  * @returns all matched files
  */
 export async function collectFiles(
     sourcePaths: string[],
     workspaceFolders: CurrentWsFolders,
-    respectGitIgnore: boolean = true,
-    maxSize = 200 * 1024 * 1024, // 200 MB
-    defaultExcludePatterns: boolean = true
-): Promise<
-    {
-        workspaceFolder: vscode.WorkspaceFolder
-        relativeFilePath: string
-        fileUri: vscode.Uri
-        fileContent: string
-        zipFilePath: string
-    }[]
-> {
-    const storage: Awaited<ReturnType<typeof collectFiles>> = []
+    options?: {
+        maxSizeLimitInBytes?: number // 200 MB default
+        useGitIgnoreFileAsFilter?: boolean // default true
+        extraExcludeFilePatterns?: string[] // default DefaultExtraExcludePatterns
+        extraFileFilterFn?: CollectFilesFileFilter
```

Comment:
what purpose is the word "extra" serving here?

```suggestion
        excludePatterns?: string[] // default DefaultExtraExcludePatterns
        filterFn?: CollectFilesFileFilter
```